lint-rules/src/main/java/com/duckduckgo/lint/strings/MissingInstructionDetector.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.lint.strings
+
+import com.android.SdkConstants
+import com.android.ide.common.resources.usage.ResourceUsageModel
+import com.android.tools.lint.detector.api.Category.Companion.CUSTOM_LINT_CHECKS
+import com.android.tools.lint.detector.api.Context
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.ResourceXmlDetector
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.XmlContext
+import com.android.tools.lint.detector.api.XmlScanner
+import org.w3c.dom.Attr
+import org.w3c.dom.Document
+import org.w3c.dom.Node
+
+@Suppress("UnstableApiUsage")
+class MissingInstructionDetector : ResourceXmlDetector(), XmlScanner {
+
+    private fun ignoreFile(context: Context) =
+        context.file.name.startsWith("donottranslate") ||
+            ResourceUsageModel.isAnalyticsFile(context.file) ||
+            !context.project.reportIssues
+
+
+    override fun visitDocument(context: XmlContext, document: Document) {
+
+        if (ignoreFile(context)) {
+            return
+        }
+
+        val root = document.documentElement
+        val translatable: Attr? = root?.getAttributeNode(SdkConstants.ATTR_TRANSLATABLE)
+        if (translatable?.value != SdkConstants.VALUE_FALSE) {
+            if (context.file.parentFile?.name == SdkConstants.FD_RES_VALUES) {
+                visitNode(context, document)
+            }
+            return
+        }
+
+    }
+
+    private fun visitNode(context: XmlContext, node: Node) {
+        val nodeType = node.nodeType
+        if (nodeType == Node.TEXT_NODE) {
+            val text = node.nodeValue
+            val instruction = node.parentNode.attributes.getNamedItem("instruction")
```

Comment:
as discussed in https://github.com/duckduckgo/Android/pull/2398/files#r996270777, can you add a check for the annotation?

Reply:
I don't really understand what you mean? Which annotation? The discussion there is about the fact that a string inside a strings.xml file (or strings-vpn.xml in that case) must be translated or have a `translatable=false` to pass lint. 

This PR is to trigger a lint error if a string with placeholders doesn't have instructions but it ignores strings in donottranslate.xml and strings with `translatable=false`

Reply:
I'm an idiot and I'll see myself out